scripts/cdist

```diff
@@ -101,6 +130,21 @@ def commandline():
         logging.root.setLevel(logging.INFO)
     if args.debug:
         logging.root.setLevel(logging.DEBUG)
+    args_dict = vars(args)
+    # if command with remote_copy and remote_exec params
+    if 'remote_copy' in args_dict and 'remote_exec' in args_dict:
+        # if remote-exec and/or remote-copy args are None then user
+        # didn't specify command line options nor env vars:
+        # inspect multiplexing options for default cdist.REMOTE_COPY/EXEC
+        if args_dict['remote_copy'] is None or args_dict['remote_exec'] is None:
+            control_path_dir = tempfile.mkdtemp(prefix="cdist.control.path")
+            import atexit
+            atexit.register(lambda: shutil.rmtree(control_path_dir))
+            mux_opts = inspect_ssh_mux_opts(control_path_dir)
+            if args_dict['remote_exec'] is None:
+                args.remote_exec = cdist.REMOTE_EXEC + mux_opts
+            if args_dict['remote_copy'] is None:
+                args.remote_copy = cdist.REMOTE_COPY + mux_opts
```

Comment:
I was wondering - can't we directly save it into the tmp directory that cdist later creates? in that case the mux socket will automatically vanish once cdist is done.

I know that from scripts/cdist the cdist object is (not yet) available, but it sounds a bit more clean to me.